subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -71,6 +74,11 @@ public WatchingVirtualFileSystem(
         this.locationsWrittenByCurrentBuild = locationsWrittenByCurrentBuild;
     }
 
+    @Override
+    public boolean isWatchingFileSystem() {
+        return watchRegistry != null;
```

Comment:
Producing the right flag for build scans depends on the order in which things happen. So we need to start the watching before we query this value for build scans. I would prefer doing the decision based on the value of the flag and whether or not the VirtualFileSystem implementation supports watching. That would mean:
- (DEFAULT || ON) & supports watching -> fs watching true
- OFF & supports watching -> fs watching false
- ON & not supports watching -> fail or off
- (DEFAULT || OFF) & not supports watching -> off

Reply:
I think your point about the ordering is valid. I think we should fix it differently, though. I moved around the emitting of the FSW state, so now the ordering is completely explicit.